lib/render/user.php

```diff
@@ -23,23 +23,32 @@ function GetUserAndTooltipDiv(
 
         $userSanitized = $user;
         sanitize_outputs($userSanitized);
+
         return '<del>' . $userSanitized . '</del>';
     }
 
-    return _GetUserAndTooltipDiv($user, $userCardInfo, $imageInstead, $customLink, $iconSizeDisplayable, $iconClassDisplayable);
+    $linkURL = '/user/' . $userCardInfo['User'];
+    if (!empty($customLink)) {
+        $linkURL = $customLink;
+    }
+
+    $displayable = $userCardInfo['User'];
+    if ($imageInstead) {
+        $displayable = "<img loading='lazy' src='" . media_asset('/UserPic/' . $user . '.png') . "' width='$iconSizeDisplayable' height='$iconSizeDisplayable' alt='$displayable' class='$iconClassDisplayable' />";
+    }
+
+    return "<span class='inline' onmouseover=\"Tip(loadCard('user', '$user'))\" onmouseout=\"UnTip()\">" .
```

Comment:
This successfully displaces the rank query, which is the more expensive part of building the tooltip, but `getUserCardData` is still called on line 17 every time this method is called (once for image, once for text, repeated for each occurrence on the page). That's one of the things I was addressing in #1130.

From viewing an achievement:
* 1130: `148 statements were executed, 5 of which were duplicated, 143 unique 11.49s`
* this: `148 statements were executed, 115 of which were duplicated, 33 unique 9.75s`

I offloaded 115 duplicate queries (redundant between image and text), but replaced them with 115 queries for the user ranks, which was clearly slower.

I'd like to reapply that logic in a separate PR after this is merged.

Reply:
I was about to add caching to `getUserCardData` - but i found it to be a very simple query only fetching account details.
Added ephemeral/per-request caching; does this address the issue? https://github.com/RetroAchievements/RAWeb/pull/1142/commits/4670b78fa17eef6bc9f3dcae114a55c47dcdfe5f

Reply:
Those numbers were from a random achievement page - 82 unique people earned the achievement (50 shown on the page), and 5 users in my all-time followed users ranking.

I tried again with and without the per-request caching, and there's clearly an impact on the number of queries, but no noticeable impact to the overall time elapsed.

* before: `148 statements were executed, 115 of which were duplicated, 33 unique 14.21s`
* after: `93 statements were executed, 5 of which were duplicated, 88 unique 14.36s`

And the math works out. 148-93=55. 55x2=110. 115-110=5. 55 duplicated statements were eliminated. Since they were each duplicated, they were counted twice in the duplicated list. Now they appear in the unique list once. 50 winners + 5 friends = 55.

However, applying the same logic to a code notes page where there were 65 code notes all from the same author:

* before: `75 statements were executed, 67 of which were duplicated, 8 unique 468ms`
* after: `11 statements were executed, 2 of which were duplicated, 9 unique 44.42ms`

The 65 duplicated lookups for the code note author became 1 unique lookup and saved 90% of the page load.

And the "recent forum posts" page (where 15 unique users posted 25 messages):

* before: `59 statements were executed, 52 of which were duplicated, 7 unique 120ms`
* after: `24 statements were executed, 2 of which were duplicated, 22 unique 98.35ms`